internal/resources/tanzukubernetescluster/resource_tanzu_kuberenetes_cluster.go

```diff
@@ -0,0 +1,424 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package tanzukubernetescluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	openapiv3 "github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper/openapi_v3_schema_validator"
+	clusterclassmodels "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/clusterclass"
+	tanzukubernetesclustermodels "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/tanzukubernetescluster"
+	tkccommon "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/tanzukubernetescluster/common"
+)
+
+func ResourceTanzuKubernetesCluster() *schema.Resource {
+	return &schema.Resource{
+		Schema:        tanzuKubernetesClusterSchema,
+		CreateContext: resourceTanzuKubernetesClusterCreate,
+		ReadContext:   resourceTanzuKubernetesClusterRead,
+		UpdateContext: resourceTanzuKubernetesClusterUpdate,
+		DeleteContext: resourceTanzuKubernetesClusterDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTanzuKubernetesClusterImporter,
+		},
+		CustomizeDiff: validateSchema,
+	}
+}
+
+// Resource functions.
+
+func resourceTanzuKubernetesClusterCreate(ctx context.Context, data *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+	model, err := tfModelResourceConverter.ConvertTFSchemaToAPIModel(data, []string{})
+
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "Couldn't create UTKG Cluster."))
+	}
+
+	request := &tanzukubernetesclustermodels.VmwareTanzuManageV1alpha1ManagementClusterProvisionerTanzukubernetesClusterData{
+		TanzuKubernetesCluster: model,
+	}
+
+	_, err = config.TMCConnection.TanzuKubernetesClusterResourceService.TanzuKubernetesClusterResourceServiceCreate(request)
```

Comment:
FYI you can refer akscluster ekscluster  resource implementation

Reply:
but the cluster API handles nodepools and as you mentioned previously on the design doc, we are going with a single resource for both cluster and nodepool and not separating them. why do we need to add the complexity of an additional client when the Cluster API provides all needed capabilities and no additional resource is being created to manage nodepools separately? this seems like over kill for this task unless i am missing something.

Reply:
Yes Ramya,  

> FYI you can refer akscluster ekscluster resource implementation

Reply:
> but the cluster API handles nodepools and as you mentioned previously on the design doc, we are going with a single resource for both cluster and nodepool and not separating them. why do we need to add the complexity of an additional client when the Cluster API provides all needed capabilities and no additional resource is being created to manage nodepools separately? this seems like over kill for this task unless i am missing something.

Today it looks like overkill because you see that Cluster API is able to handle nodepool operations but it will not work in the long run because the LCM team roadmap is to stop supporting nodepool operations via cluster API and only allow nodepool operations via nodepool APis . Thats why you see those 2 options of whether to have 2 resources(cluster and nodepool)/single resource(cluster including nodepool) is documented.